subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -334,31 +337,34 @@ void radio_pkt_configure(uint8_t bits_len, uint8_t max_len, uint8_t flags)
 	/* To use same Data Channel PDU structure with nRF5 specific overhead
 	 * byte, include the S1 field in radio packet configuration.
 	 */
-	if (dc) {
+	if (pdu_type == RADIO_PKT_CONF_PDU_TYPE_DC) {
 		extra |= (RADIO_PCNF0_S1INCL_Include <<
 			  RADIO_PCNF0_S1INCL_Pos) & RADIO_PCNF0_S1INCL_Msk;
+#if defined(CONFIG_BT_CTLR_DF)
+		if (RADIO_PKT_CONF_CTE_GET(flags) == RADIO_PKT_CONF_CTE_ENABLED) {
+			bits_s1 = 8U;
+		} else
+#endif /* CONFIG_BT_CTLR_DF */
+		{
+			bits_s1 = 0U;
+		}
+	} else {
+		bits_s1 = 0U;
 	}
 #endif /* CONFIG_SOC_COMPATIBLE_NRF52X */
 
-	NRF_RADIO->PCNF0 = (((1UL) << RADIO_PCNF0_S0LEN_Pos) &
-			    RADIO_PCNF0_S0LEN_Msk) |
-			   ((((uint32_t)bits_len) << RADIO_PCNF0_LFLEN_Pos) &
-			    RADIO_PCNF0_LFLEN_Msk) |
-			   ((((uint32_t)8-bits_len) << RADIO_PCNF0_S1LEN_Pos) &
-			    RADIO_PCNF0_S1LEN_Msk) |
-			   extra;
+	NRF_RADIO->PCNF0 =
+		(((1UL) << RADIO_PCNF0_S0LEN_Pos) & RADIO_PCNF0_S0LEN_Msk) |
+		((((uint32_t)bits_len) << RADIO_PCNF0_LFLEN_Pos) & RADIO_PCNF0_LFLEN_Msk) |
+		((((uint32_t)bits_s1) << RADIO_PCNF0_S1LEN_Pos) & RADIO_PCNF0_S1LEN_Msk) | extra;
 
 	NRF_RADIO->PCNF1 &= ~(RADIO_PCNF1_MAXLEN_Msk | RADIO_PCNF1_STATLEN_Msk |
 			      RADIO_PCNF1_BALEN_Msk | RADIO_PCNF1_ENDIAN_Msk);
-	NRF_RADIO->PCNF1 |= ((((uint32_t)max_len) << RADIO_PCNF1_MAXLEN_Pos) &
-			     RADIO_PCNF1_MAXLEN_Msk) |
-			    (((0UL) << RADIO_PCNF1_STATLEN_Pos) &
-			     RADIO_PCNF1_STATLEN_Msk) |
-			    (((3UL) << RADIO_PCNF1_BALEN_Pos) &
-			     RADIO_PCNF1_BALEN_Msk) |
-			    (((RADIO_PCNF1_ENDIAN_Little) <<
-			      RADIO_PCNF1_ENDIAN_Pos) &
-			     RADIO_PCNF1_ENDIAN_Msk);
+	NRF_RADIO->PCNF1 |=
+		((((uint32_t)max_len) << RADIO_PCNF1_MAXLEN_Pos) & RADIO_PCNF1_MAXLEN_Msk) |
+		(((0UL) << RADIO_PCNF1_STATLEN_Pos) & RADIO_PCNF1_STATLEN_Msk) |
```

Comment:
Not sure if these lines were actually changed, or if you just formatted them, but isn't it redundant to do `<<` on a 0 value?

Reply:
It was just re-formatted. I think that it is just for sake of readability, to make the configuration more explicit even though it is redundant. Compiler will optimize it anyway.